### PR TITLE
Kovri: fixes + updates

### PR DESCRIPTION
**Pull request by anonimal**
This is a first round of fixes for the Kovri script. RFC.

If it's too verbose or technical, let's see what else we can come up with.

### CODE REVIEW

File: kovri.md [hunk 1]
```diff
@@ -1,64 +1,68 @@
...
 
 Until this point, this video series has covered how Monero obfuscates information stored on the blockchain.
 Ring signatures obscure the sender.
-Stealth addressing prevents anyone but the sender and the recipient from knowing the recipient owns Monero.
-RingCT hides the amount of Monero transmitted.
-However, some personally identifiable information can still be leaked in the process of sending Monero.
-This video will focus on a technology that closes that gap, called Kovri.
+Stealth addressing prevents anyone, other than the sender and the recipient, from knowing that the recipient owns Monero.
+RingCT hides the amount of Monero that's transmitted.
+However, some personally identifiable information will still be leaked in the process of sending Monero.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: "However, some personally identifiable information will still be leaked when making a transaction."

File: kovri.md [hunk 2]
```diff
@@ -1,64 +1,68 @@
...
+This video focuses on the technology that closes this gap of privacy. This technology is called Kovri.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: I think I prefer this as a single sentence.

File: kovri.md [hunk 3]
```diff
@@ -1,64 +1,68 @@
...
 
 Suppose Alice wants to send a transaction to Bob.
 She generates a transaction on her local device, and gets ready to broadcast it to the network, so it can be included in the next block.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: "She generates a transaction on her local device and gets ready to broadcast it to the network, so it can be included in the next block."

File: kovri.md [hunk 4]
```diff
@@ -1,64 +1,68 @@
...
-Alice is connected to only a small portion of the nodes on the Monero network. Her transaction request will need to reach as many nodes as possible - ideally all of them - for her to have the greatest chance of this transaction getting included in the next block.
-Let's say she is connected to five other nodes, so she broadcasts her transaction to all five.
-These five nodes then forward her transaction to all the nodes they are connected to, and so on, until many (or all) nodes have this transaction.
+Alice is connected to only a small portion of the nodes on the Monero network. Her transaction request will need to reach as many nodes as possible - ideally all of them - for her to have the greatest chance of this transaction to be included in the next block.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: "Alice only connects to a small portion of the nodes on the Monero network. Her transaction request will need to reach as many nodes as possible - ideally all of them - for her transaction to have the greatest chance of being included in the next block."

File: kovri.md [hunk 5]
```diff
@@ -1,64 +1,68 @@
...
+Let's say she connects to five other nodes. We'll call each node "Bob". Once connected, she broadcasts her transaction to all five Bobs.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: Perhaps remove these second and third sentences. We can just remove all references to Bob entirely.

File: kovri.md [hunk 6]
```diff
@@ -1,64 +1,68 @@
...
+These Bobs then forward her transaction to all the nodes they are connected to, and so on, until many (or all) nodes have this transaction.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: Maybe "these nodes"

File: kovri.md [hunk 7]
```diff
@@ -1,64 +1,68 @@
...
 
-Unfortunately, Alice takes a risk to broadcast her transaction.
-Some of the nodes on the network may be logging the IP addresses that they get transaction requests from.
+Unfortunately, Alice takes a risk when broadcasting her transaction.
+Some of the nodes on the network may be logging the IP addresses they get when transaction requests are given.
 An IP address is like your home address for your internet connection.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: "An IP address is like a home address for the internet."

File: kovri.md [hunk 8]
```diff
@@ -1,64 +1,68 @@
...
 Attackers will try to associate transactions with the IP addresses they originate from.
 They do not know anything about what address a transaction is coming from, where it is going to, or what the amount is.
 
-However, an attacker with enough resources could knock on Alice's door and demand that she tell them this information. 
+However, an attacker with enough resources could knock on Alice's door and demand that she tell them this information.
 Alternatively, they could purposefully attempt to block her transaction from being broadcast to the rest of the network.
 
 So what can Alice do to mitigate these threats?
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: Add at the end "Well, she can use Kovri!"

File: kovri.md [hunk 9]
```diff
@@ -1,64 +1,68 @@
...
 
-Alice does not know which nodes log her IP address, so she can't simply avoid the malicious nodes.
-Her only choice is to hide her IP address.
-There are several ways to hide an IP address, including with Tor, a trusted VPN, or I2P.
-We will focus on I2P in today's video, since this is the network on which Kovri operates.
+Alice does not know which Bob logs her IP address, so she can't simply avoid the malicious Bobs.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: I suggest removing lines 27-39. It's good content, but perhaps is not best for this video.

File: kovri.md [hunk 10]
```diff
@@ -1,64 +1,68 @@
...
+Trusted VPNs also don't protect Bob from Alice knowing who Bob is.
+
+It's all so complicated! Fortunately, Alice and Bob have a better option for protecting their privacy. Their option is Kovri.
 
-In a sentence, Kovri is a lightweight router compatible with cryptocurrencies and built for the I2P network.
-I2P stands for Invisible Internet Protocol.
-I2P is a layer of the internet that allows for all connections in the network to be private.
-In contrast with regular internet routing or TOR's onion routing, I2P utilizes what it refers to as garlic routing.
-Instead of connecting with participants directly or through TOR nodes, such as going directly to a website, all information is encrypted and relayed through other I2P routers in the network.
+Kovri is a free, decentralized, anonymity technology developed by Monero.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: "Kovri is a free, decentralized anonymity technology developed by Monero."

Remove the comma

File: kovri.md [hunk 11]
```diff
@@ -1,64 +1,68 @@
...
+Currently based on the Invisible Internet Project's open specifications, Kovri uses both garlic encryption and garlic routing to create a private, protected overlay-network across the internet.
+This overlay-network provides Alice and Bob with the ability to effectively hide their geographical location and internet IP address.
+Essentially, Kovri *covers* an application's internet traffic to make it anonymous within the network.
+A lightweight and security-focused router, Kovri is fully compatible with the I2P network.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: "Kovri is lightweight, security-focused, and fully compatible with the I2P network"

File: kovri.md [hunk 12]
```diff
@@ -1,64 +1,68 @@
...
 
-Since I2P routers know little or no information about the data they are sending, where it originated, or its destination, all information in this network is private and anonymous.
-If Alice sends her transaction with Kovri, nodes that log Monero traffic will no longer get her real IP address, rendering these passive surveillance tactics useless.
+If Alice sends her transaction with Kovri, nodes that log Monero traffic will no longer see her internet IP address, rendering passive surveillance tactics close to useless.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: "With Kovri, passive surveillance attacks on Monero are close to useless"

File: kovri.md [hunk 13]
```diff
@@ -1,64 +1,68 @@
...
 
 Now let’s imagine a different scenario.
-Suppose Alice has an internet service provider that does not want her to use Monero, who send her a warning saying so.
-Alice can optionally send all the Monero sync data, not just the transaction broadcast, through I2P.
+Suppose Alice has an internet service provider that does not want her to use Monero, and send her a warning to cease and desist.
+Alice can optionally send all the Monero sync data, not just the transaction broadcast, through Kovri.
 Now, her Monero connections are hidden from her ISP.
 This improves Monero's censorship resistance, since other people no longer know who is running a Monero node and demand that they be shut down.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: "This improves Monero's censorship resistance, since other people no longer know who is running a Monero node, and others can not demand that they be shut down."

File: kovri.md [hunk 14]
```diff
@@ -1,64 +1,68 @@
...
+However, if Charlie can send his solved blocks through Kovri, then no one can censor the work from his pool, since they no longer know that his pool created it.
 
-Kovri will come bundled with future Monero releases, and it will be enabled by default to broadcast the transaction through the I2P network.
-It will feature a common API that can be used by any cryptocurrency, not just Monero.
+Kovri will come bundled with future Monero releases, and it will be enabled by default to broadcast the transaction anonymously.
+Kovri will feature a common API that can be used by any cryptocurrency, not just Monero.
 
-In summary, Kovri is a software router that allows Monero users to connect to the private I2P network to send transactions anonymously.
-Malicious nodes can no longer threaten users who create these transactions or block them from propagating across the network. 
+In summary, Kovri is an anonymizing router software that allows Monero users to send transactions anonymously while enjoying the benefits of the I2P network.
```
Comment (SamsungGalaxyPlayer) [inline kovri.md]: "In summary, Kovri is an anonymizing router software that allows Monero users to send transactions anonymously with less privacy and security concerns."

**Review by SamsungGalaxyPlayer — CHANGES_REQUESTED** [review]

File: kovri.md (hunk 1 above)
Comment (anonimal) [reply kovri.md]: Agreed.

File: kovri.md (hunk 2 above)
Comment (anonimal) [reply kovri.md]: >This video will focus on a technology that closes that gap, called Kovri.

Prior to the revision, even with the comma, this sentence too closely implied that Kovri was that gap.

Let's try a 3rd revision.

File: kovri.md (hunk 3 above)
Comment (anonimal) [reply kovri.md]: Agreed.

File: kovri.md (hunk 4 above)
Comment (anonimal) [reply kovri.md]: Agreed.

File: kovri.md (hunk 5 above)
Comment (anonimal) [reply kovri.md]: >We can just remove all references to Bob entirely

Bob is factually important because Alice doesn't directly connect to every node on the network. We shouldn't remove Bob if we talk about Alice and Charlie (actual crypto names).

Perhaps we should rephrase instead. I'll play with this.

File: kovri.md (hunk 6 above)
Comment (anonimal) [reply kovri.md]: Agreed.

File: kovri.md (hunk 7 above)
Comment (anonimal) [reply kovri.md]: Agreed, good enough.

File: kovri.md (hunk 8 above)
Comment (anonimal) [reply kovri.md]: Great touch.

File: kovri.md (hunk 9 above)
Comment (anonimal) [reply kovri.md]: Agreed. I can put it in the kovri FAQ or similar.

File: kovri.md (hunk 10 above)
Comment (anonimal) [reply kovri.md]: Actually, this is perfectly valid English and does not reflect on the spoken language. If the reader is pedantic, he may interpret otherwise. And that's good if he does, we need the space for breath.

File: kovri.md (hunk 11 above)
Comment (anonimal) [reply kovri.md]: Agreed.

File: kovri.md (hunk 12 above)
Comment (anonimal) [reply kovri.md]: Well, if we're going to pick this apart, we should be clear that all surveillance attacks on overlay networks are still useful but become more difficult depending on the network.

File: kovri.md (hunk 13 above)
Comment (anonimal) [reply kovri.md]: Agreed.

File: kovri.md (hunk 14 above)
Comment (anonimal) [reply kovri.md]: Meh, I don't care for this revision. Let's come up with a 3rd that's better than our previous two.

**Comment by anonimal** [comment]
Revised, rebased. Let's do another round of review.

**Comment by ajs-xmr** [comment]
Will give it a once over this weekend

File: kovri.md (hunk 2 above)
Comment (SamsungGalaxyPlayer) [reply kovri.md]: I see your point. Good catch.
